src/main/java/net/fabricmc/loader/impl/ObjectShareImpl.java

```diff
@@ -80,7 +90,7 @@ public synchronized void whenAvailable(String key, BiConsumer<String, Object> co
 		if (value != null) {
 			consumer.accept(key, value);
```

Comment:
This is still a callback in a synchronized block. :-)
I changed it to be this:
````
	@Override
	public void whenAvailable(String key, BiConsumer<String, Object> consumer) {
		validateKey(key);

		Object value = null;
		synchronized (this) {
			value = get(key);
			if (value == null) {
				pending.computeIfAbsent(key, ignore -> new ArrayList<>()).add(consumer);
			}
		}

		if (value != null) {
			consumer.accept(key, value);
		}
	}
````

Reply:
it's clearly there? ;)